File: StackImplementation/app/src/main/java/com/srinivas/stackimplementation/CharacterStack.java
package com.srinivas.stackimplementation;

public class CharacterStack {
    static int maxSize;
    char stack[] = new char[maxSize];
    int currentIndex = -1;

    static {
        maxSize = (int) Math.pow(10, 4) / 2;
    }

    void push(char value) {
        if (currentIndex == maxSize) {
            return;
        } else {
            currentIndex += 1;
            stack[currentIndex] = value;
        }
    }

    void pop() {
        if (currentIndex < 0) {
            return;
        } else {
            currentIndex -= 1;
        }
    }

    Character top() {
        if (currentIndex < 0) {
            return null;
        } else {
            return stack[currentIndex];
        }
    }



    public static void main(String [] args){

    }
}

File: StackImplementation/app/src/main/java/com/srinivas/stackimplementation/CharacterStack.kt
package com.srinivas.stackimplementation

import kotlin.math.pow

class CharacterStack {
    var stack = CharArray(maxSize)
    var currentIndex = -1

    companion object {
        var maxSize = 0

        init {
            maxSize = 10.0.pow(4.0).toInt() / 2
        }
    }

    fun push(value: Char) {
        if (currentIndex == maxSize) {
            return
        } else {
            currentIndex += 1
            stack[currentIndex] = value
        }
    }

    fun pop() {
        currentIndex -= if (currentIndex < 0) {
            return
        } else {
            1
        }
    }

    fun top(): Char? {
        return if (currentIndex < 0) {
            null
        } else {
            stack[currentIndex]
        }
    }
}